exporters/ostream/include/opentelemetry/exporters/ostream/span_exporter.h

```diff
@@ -0,0 +1,148 @@
+#pragma once
+
+#include "opentelemetry/sdk/trace/exporter.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+#include "opentelemetry/nostd/type_traits.h"
+#include <opentelemetry/version.h>
+
+#include <sstream>
+#include <iostream>
+#include <map>
+
+namespace nostd    = opentelemetry::nostd;
+namespace sdktrace = opentelemetry::sdk::trace;
+
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace exporter
+{
+namespace trace
+{
+
+/**
+ * The OStreamSpanExporter exports span data through an ostream
+*/
+class OStreamSpanExporter final : public sdktrace::SpanExporter
+{
+public:
+
+/**
+ * Create a OStreamSpanExporter. This constructor takes in a reference to an ostream that the
+ * export() function will send span data into.
+ * The default ostream is set to stdout
+ */
+  explicit OStreamSpanExporter(std::ostream &sout = std::cout) noexcept;
+
+  std::unique_ptr<sdktrace::Recordable> MakeRecordable() noexcept override;
+
+  sdktrace::ExportResult Export(
+      const nostd::span<std::unique_ptr<sdktrace::Recordable>> &spans) noexcept override;
+
+  void Shutdown(std::chrono::microseconds timeout = std::chrono::microseconds(0)) noexcept override;
+
+private:
+  std::ostream &sout_;
+  bool isShutdown_ = false;
+
+  // Mapping status number to the string from api/include/opentelemetry/trace/canonical_code.h
+  std::map<int, std::string> statusMap {
+    {0, "OK"},
+    {1, "CANCELLED"},
+    {2, "UNKNOWN"},
+    {3, "INVALID_ARGUMENT"},
+    {4, "DEADLINE_EXCEEDED"},
+    {5, "NOT_FOUND"},
+    {6, "ALREADY_EXISTS"},
+    {7, "PERMISSION_DENIED"},
+    {8, "RESOURCE_EXHAUSTED"},
+    {9, "FAILED_PRECONDITION"},
+    {10,"ABORTED"},
```

Comment:
nit: would you put a space here (if you like vertical alignment, put two spaces for single digit).